nosnoc/problem.py

```diff
@@ -504,21 +503,19 @@ def forward_simulation(self, ocp: NosnocOcp, Uk: ca.SX, sot: ca.SX) -> None:
 
         for j in range(opts.n_s):
             # Dynamics excluding complementarities
-            p_stage = self.p[:4]
-            fj = sot * model.f_x_fun(X_fe[j], self.rk_stage_z(j), Uk, self.p_stage, model.v_global)
+        
+            fj = sot * model.f_x_fun(X_fe[j], self.rk_stage_z(j), Uk, self.p, model.v_global)
             qj = sot * ocp.f_q_fun(X_fe[j], Uk, self.p, model.v_global)
             gj = model.g_z_all_fun(X_fe[j], self.rk_stage_z(j), Uk, self.p)
             self.add_constraint(gj)
             if opts.dcs_mode == DcsMode.PDS:
                 # Add PDS-specific constraints
-                # lam = self.Lambda(stage=j)  # or use self.ind_lam
                 lam = self.w[flatten(self.ind_lam[j])]
-                # c_pds is a list, one per system
-                for sys_idx in range(model.dims.n_sys):
-                    c_pds_val = ca.Function('c_pds_tmp', [model.x], [model.c_pds[sys_idx]])(X_fe[j])  # or however you evaluate c_pds
-                    self.add_constraint(c_pds_val - lam[sys_idx])
-                # Add the unconstrained dynamics as the ODE
-                self.add_constraint(fj - ca.Function('f_unconstrained_tmp', [model.x], [model.f_unconstrained[sys_idx]])(X_fe[j]))
+                c_pds_val = model.c_pds_fun(X_fe[j])
+                self.add_constraint(c_pds_val - lam)
```

Comment:
This would imply that $\lambda = c(x)$ which is not quite right. All you need here is actually the lower bound: $c(x) \ge 0$